src/ray/object_manager/object_manager.cc

```diff
@@ -177,20 +177,29 @@ ray::Status ObjectManager::PullEstablishConnection(const ObjectID &object_id,
   RAY_CHECK_OK(status);
 
   if (conn == nullptr) {
-    status = object_directory_->GetInformation(
-        client_id,
-        [this, object_id, client_id](const RemoteConnectionInfo &connection_info) {
-          std::shared_ptr<SenderConnection> async_conn = CreateSenderConnection(
-              ConnectionPool::ConnectionType::MESSAGE, connection_info);
-          connection_pool_.RegisterSender(ConnectionPool::ConnectionType::MESSAGE,
-                                          client_id, async_conn);
-          Status pull_send_status = PullSendRequest(object_id, async_conn);
-          RAY_CHECK_OK(pull_send_status);
-        },
-        [](const Status &status) {
-          RAY_LOG(ERROR) << "Failed to establish connection with remote object manager.";
-          RAY_CHECK_OK(status);
-        });
+    if (config_.max_sender_connection_count == 0 ||
+        connection_pool_.CountSender(ConnectionPool::ConnectionType::MESSAGE, client_id) <
+            config_.max_sender_connection_count) {
```

Comment:
Create a function to judge whether remain available connection or not, instead of countSender function